### ISSUE TITLE
Incorrect autocorrect for `Style/AccessModifierDeclarations` cop

### ISSUE THREAD

**Issue by nerixim** [rubocop/rubocop#13433]
After updating from 1.65.1 to 1.68.0, Style/AccessModifierDeclarations started to having false positives for `module_function`.

--------

## Expected behavior

No offenses for the following code:

```ruby
module Foo
  def bar; end
  def baz; end

  module_function :bar, :baz
end

```

## Actual behavior

```shell
> rubocop --debug lib/foo.rb
For /Users/user/code: configuration from /Users/user/code/.rubocop.yml
configuration from /Users/user/.local/share/mise/installs/ruby/3.2.5/lib/ruby/gems/3.2.0/gems/rubocop-rspec-3.0.3/config/default.yml
configuration from /Users/user/.local/share/mise/installs/ruby/3.2.5/lib/ruby/gems/3.2.0/gems/rubocop-rspec-3.0.3/config/default.yml
Default configuration from /Users/user/.local/share/mise/installs/ruby/3.2.5/lib/ruby/gems/3.2.0/gems/rubocop-1.68.0/config/default.yml
configuration from /Users/user/.local/share/mise/installs/ruby/3.2.5/lib/ruby/gems/3.2.0/gems/rubocop-graphql-1.5.3/config/default.yml
configuration from #{path}
configuration from /Users/user/.local/share/mise/installs/ruby/3.2.5/lib/ruby/gems/3.2.0/gems/rubocop-rspec_rails-2.30.0/config/default.yml
configuration from /Users/user/.local/share/mise/installs/ruby/3.2.5/lib/ruby/gems/3.2.0/gems/rubocop-rspec_rails-2.30.0/lib/../config/default.yml
configuration from /Users/user/.local/share/mise/installs/ruby/3.2.5/lib/ruby/gems/3.2.0/gems/rubocop-factory_bot-2.26.1/config/default.yml
configuration from /Users/user/.local/share/mise/installs/ruby/3.2.5/lib/ruby/gems/3.2.0/gems/rubocop-factory_bot-2.26.1/lib/../config/default.yml
Inheriting configuration from /Users/user/code/.rubocop.base.yml
Inheriting configuration from /Users/user/code/.rubocop.rails.yml
configuration from /Users/user/.local/share/mise/installs/ruby/3.2.5/lib/ruby/gems/3.2.0/gems/rubocop-rails-2.25.1/config/default.yml
configuration from /Users/user/.local/share/mise/installs/ruby/3.2.5/lib/ruby/gems/3.2.0/gems/rubocop-rails-2.25.1/config/default.yml
Inheriting configuration from /Users/user/code/.rubocop_todo.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Users/user/code/lib/foo.rb
C

Offenses:

lib/foo.rb:5:3: C: [Correctable] Style/AccessModifierDeclarations: module_function should not be inlined in method definitions.
  module_function :bar, :baz
  ^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
Finished in 0.47796599997673184 seconds
```

## Steps to reproduce the problem

Update rubocop from 1.65.1 to 1.68.0.

## RuboCop version

```
$ [bundle exec] rubocop -V
1.68.0 (using Parser 3.3.6.0, rubocop-ast 1.34.1, analyzing as Ruby 3.2, running on ruby 3.2.5) [arm64-darwin23]
  - rubocop-factory_bot 2.26.1
  - rubocop-graphql 1.5.3
  - rubocop-rails 2.25.1
  - rubocop-rspec 3.0.3
  - rubocop-rspec_rails 2.30.0
```

**Comment by dvandersluis** [comment]
Can you share your config for this cop?

**Comment by dvandersluis** [comment]
`module_function` is an access modifier like `public`, `private` and `protected`. There was a bug where inlined modifiers weren't being caught when there was more than one symbol, which was fixed in #13404. Previously, `module_function :bar` would register an offense, but `module_function :bar, :baz` wouldn't, which is why you'd suddenly get an offense from it. So this appears to be working as intended.

**Comment by nerixim** [comment]
@dvandersluis 
Thanks for clarification. I was kinda aware that there were some change regarding `module_function`.

Sorry I didn't mention that in my initial post, but autocorrect is wrong for multiple arguments. Is that a known bug/feature too?

For the original example, I get the following autocorrection:
```ruby
# original
module Foo
  def bar; end
  def baz; end

  module_function :bar, :baz
end

# corrected
module Foo
  def bar; end
  def baz; end

  :bar
end
```

**Comment by paduszynski-sixeye** [comment]
I agree autocorrection is broken right now for it

**Comment by dvandersluis** [comment]
@nerixim I cannot replicate the correction you are getting, can you share your configuration for `Style/AccessModifierDeclarations`? I get this instead:

```ruby
module Foo
  def baz; end

module_function

def bar; end
end
```

That being said, that's not autocorrecting properly, so I'm going to reopen this and fix it.

**Comment by dvandersluis** [comment]
I opened #13443.
